### DISCUSSION TITLE
nuxt-link: Why is `isActive` false?

### DISCUSSION THREAD

**Discussion by ahoiroman** [nuxt/nuxt#16194] [Questions] [answered]
Hello,

I got these pages in my */pages* directory:

```
pages
├── auth
│   └── signin.vue
├── company
│   ├── [id].vue
│   └── index.vue
└── index.vue
```

Now I got this `nuxt-link`:

```
<nuxt-link to="/company"
           v-slot="{ isActive }">
    {{isActive ? 'Active' : 'Not active'}}
</nuxt-link>
```

If I now navigate to _http://local-site/company_, the `isActive`-flag is `true` - but if I navigate to _http://local-site/company/1_, it is `false`.

As I'd like to attach the active class to the menu item if the user navigated to a specific company, this is bugging me a bit.

I hope I don't have to use workarounds like `$route.path.startsWith()`, as it feels like a dirty workaround.

Any ideas here?

**Comment by IsraelOrtuno** [comment, marked as answer]
To make it work you need to have a nested route. I know it's a bit tricky and confusing but as of now works like that:
https://stackblitz.com/edit/github-c6ex6y?file=pages/posts.vue

You are missing a `company.vue` file in your pages structure which will likely include a simple `<NuxtPage />` component.

**Comment by ahoiroman** [comment]
Thanks. This indeed does work, but I did not find this in the docs (searching for nuxt-link), so I learned something new today.

Thank you!

**Reply by IsraelOrtuno** [reply]
Vue-router docs could help with that https://router.vuejs.org
